src/interfaces/IDynamicRandomVariableGenerator.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: BUSL 1.1
+// Valorem Labs Inc. (c) 2022
+pragma solidity 0.8.16;
+
+/**
+ * @notice Defines the interface for a generator of random variables, selecting from
+ * a range of elements, with a dynamically weighted probability distribution function. 
+ */
+interface IDynamicRandomVariableGenerator {
+
+    /*//////////////////////////////////////////////////////////////
+    //  Data Structures
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice An element that can be selected from many, with liklihood 
+    /// of selection dictated by its weight.
+    struct Element {
```

Comment:
I guess the question here is why this needs to be an element struct with a weight and index rather than just a member of an array that is a weight